lsp-headerline.el

```diff
@@ -89,15 +89,14 @@ for caching purposes.")
 
 (defun lsp-headerline--filename-with-icon (file-path)
   "Return the filename from FILE-PATH with the extension related icon."
-  (let ((filename (f-filename file-path))
-        (file-ext (f-ext file-path)))
-    (if file-ext
-        (when (require 'lsp-treemacs nil t)
-          (if-let (icon (lsp-treemacs-get-icon file-ext))
-              (format "%s %s"
-                      (lsp-headerline--fix-image-background icon)
-                      filename)
-            filename))
+  (let ((filename (f-filename file-path)))
+    (if-let ((file-ext (f-ext file-path))
```

Comment:
At least in the version of Emacs I'm using, this appears to expand to `(if-let* ...`, which is unexpected to me but makes some sense.

I think it's less confusing to write `(if-let*` instead. But since the CI requires old Emacsen, we should use `dash`'s `(-if-let*`.